src/app/tools/point_shapes.h

```diff
@@ -121,12 +140,7 @@ class BrushPointShape : public PointShape {
             if (b == maskIndex) b = 0;
             else b = loop->getPalette()->getEntry(b);
             // Same as in RGBA gradient
-            if (rgba_geta(a) == 0) a = b;
-            else if (rgba_geta(b) == 0) b = a;
-            a = doc::rgba(int(ti*rgba_getr(a) + t*rgba_getr(b)),
-                          int(ti*rgba_getg(a) + t*rgba_getg(b)),
-                          int(ti*rgba_getb(a) + t*rgba_getb(b)),
-                          int(ti*rgba_geta(a) + t*rgba_geta(b)));
+            rgbaGradient();
```

Comment:
This is the important place, here we are really passing the new updated value of b variable:
```suggestion
            a = rgbaGradient(a, b);
```